src/net/sock_posix.go

```diff
@@ -13,6 +13,17 @@ import (
 	"syscall"
 )
 
+type SockTrace struct {
+	DidRead  func(int)
+	DidWrite func(int)
+}
+
+func WithSockTrace(ctx context.Context, trace *SockTrace) context.Context {
+	return context.WithValue(ctx, sockTraceKey{}, trace)
+}
+
+type sockTraceKey struct{}
```

Comment:
so if you wanted to be lazy and bend the context.Context rules a bit, you can just use a string as the context.Value key, like:

"[tailscale.com]didread-func" and "[tailscale.com]didwrite-func" and then ditch these three symbols.

Then you could use it safely from the tailscaled side without build tags. Either way.

Reply:
Was thinking that it should be something more type safe so that 
1. we get an error if you build it with the wrong toolchain (as opposed to stats silently breaking)
2. it's a nicer API, and in theory could be upstreamed

But if you think there's ~0 chance that this could/should be upstreamed, happy to change it.